### ISSUE TITLE
Scroll position lost when hiding read posts

### ISSUE THREAD

**Issue by aaronmw** [Memmy-App/memmy#492]
**Describe the bug**
My expectation was that I'd be scrolled to the top, but I am left looking at the "middle" of a list of unseen posts.

**To Reproduce**
1. Read some posts
2. Tap the "hide read" button
3. Notice you have to scroll UP to see stuff you haven't seen before

**Expected behavior**
Scroll me to the top

**Comment by sgriff96** [comment]
I think this would be more jarring/confusing personally to be scrolled to the top every time. You already scrolled down and i assume didn't read every single post right? But you scrolled past those posts for a reason. So now you'd have to scroll past them again. Unless I'm misunderstanding something, I think current behavior makes more sense. You can also always just tap the Feed menu button again to go to the top.

**Comment by lazyjiji** [comment]
> 

What happens currently is all the “read” posts disappear, and the yet-unseen posts further down the feed end up getting sent to the top, so you have to scroll back up to now view the unseen posts. Ideally the hiding would hide the read posts and “hold” your position so the next unseen post below the currently read ones would still be just “below” your screen.

**Comment by sgriff96** [comment]
Ah I understand now. I'm doing something similar with the index of an item in a feed right now, so might be a similar problem. Will look into it 👍

**Comment by aaronmw** [comment]
I think what I'm really missing is a "hide all above this post". If I've scrolled past it, I either read it without interacting (just a headline or joke or small enough it's all there on the post card), or I ignored it. Either way, I don't want to see it again. Apollo's swipe left to hide all above solved this: read or not, I want everything above the fold gone.

As I write that out though, it feels like a separate issue. The bug in this issue still stands though.

**Comment by lazyjiji** [comment]
Yes, I personally used the “swipe to hide” for a single post in Apollo because sometimes I did want to scroll back to previous ones so “hide all” felt too much, but agreed separate issue. I was hoping once the bigger problems clear up to request a separate feature to not have to interact at all to hide post, something akin to Apollo’s swipe to hide. Because I like to keep a “clean feed” and hide everything possible, and tapping in to posts I don’t care about just so I can hide them is a bit annoying.

**Comment by aaronmw** [comment]
I'm surprised this issue isn't higher, priority-wise. Are people's feeds just full of posts that have been there for days? Once I've seen a post, I want it gone. Otherwise, I'll save it. Would not have guessed I was in a minority!

**Comment by sgriff96** [comment]
> I'm surprised this issue isn't higher, priority-wise. Are people's feeds just full of posts that have been there for days? Once I've seen a post, I want it gone. Otherwise, I'll save it. Would not have guessed I was in a minority!

This issue isn't what you just described. You just said you want a way to hide all read posts. We have that already. What you described in the original issue is not that, it's just to keep track of position in the feed.

**Comment by aaronmw** [comment]
The "Mark Read on Scroll" feature actually addresses that tangential issue I described above — a way to clear everything I've scrolled past, whether I've interacted with it or not.

Looking forward to this original issue getting resolved though! I obsessively tap that "hide" button while I'm scrolling and each time resent having to then tap "Feed" to scroll back to the top. Since I have "Mark read on scroll" turned on, it also has the annoying side-effect of marking every post above the fold as "read", even though I haven't actually seen them yet — they "scrolled past" in one frame after everything was hidden. If I accidentally tap "Hide read posts" twice, I'll be clearing posts I'd never seen. The longer I wait between clearing read posts, the more posts are erroneously marked "read" in the process ☹️
